tests/e2e/router.hash.spec.ts

```diff
@@ -0,0 +1,157 @@
+import { test, expect } from "@tests/support/fixtures.js";
+import {
+  expectBackAndForwardNavigation,
+  expectUrlPersistsReload,
+} from "@tests/support/router.js";
+
+test.beforeEach(async ({ page }) => {
+  await page.addInitScript(() => {
+    window.HASH_ROUTING = true;
+  });
+});
+
+// Check that reloading the project page doesn't mess up history
+// Verify that the URL remains the same when reloading a project view
```

Comment:
I think we don't need this comment, it's quite obvious what's happening in the test to me.

Reply:
a57d217b

Reply:
Removed plenty redundant comments, wanted to keep it very clear to the developer unknown to selectors and the page